homeassistant/components/switch/vultr.py

```diff
@@ -0,0 +1,107 @@
+"""
+Support for interacting with Vultr subscriptions.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch.vultr/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_NAME
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.components.vultr import (
+    CONF_SUBSCRIPTION, ATTR_AUTO_BACKUPS, ATTR_ALLOWED_BANDWIDTH,
+    ATTR_CREATED_AT, ATTR_SUBSCRIPTION_ID, ATTR_SUBSCRIPTION_NAME,
+    ATTR_IPV4_ADDRESS, ATTR_IPV6_ADDRESS, ATTR_MEMORY, ATTR_DISK,
+    ATTR_COST_PER_MONTH, ATTR_OS, ATTR_REGION, ATTR_VCPUS, DATA_VULTR)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Vultr {}'
+DEPENDENCIES = ['vultr']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SUBSCRIPTION): cv.string,
+    vol.Optional(CONF_NAME): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Vultr subscription switch."""
+    vultr = hass.data.get(DATA_VULTR)
+    if not vultr:
+        return False
+
+    subscription = config.get(CONF_SUBSCRIPTION)
+    name = config.get(CONF_NAME)
+
+    if subscription in vultr.data:
+        add_devices([VultrSwitch(vultr, subscription, name)], True)
+    else:
+        _LOGGER.error(
+            "Subscription %s not found. Perhaps API key issue?",
+            subscription)
+        return False
+
+
+class VultrSwitch(SwitchDevice):
+    """Representation of a Vultr subsciption switch."""
+
+    def __init__(self, vultr, subscription, name):
+        """Initialize a new Vultr switch."""
+        self._vultr = vultr
+        self._subscription = subscription
+        self.data = self._vultr.data[subscription]
```

Comment:
Dito.

Reply:
With the switch component, I use this `self.data` immediately to format the name, allowing the default name of `Vultr {}` to be populated with the subscription's label. How do you think I should handle this? I could not use the label at all and default to `Vultr subscription` or something. Could move the name setting into update method. Could set data to None, but access the `self._vultr.data[subscription]['label']` directly in this init method.

Thanks for you assistance with my PRs, it's always much appreciated and makes me proud to contribute to such a cool project.

Reply:
Now I see the idea. As far as I can tell, is it required that the user provide the right string at the moment. Only something like `MyName {}` will work. If `MyName` is set then there will be an IndexError. My guess is that something like this could work:

```python
    [...]
     self._name = name

@property
def name(self):
"""Return the name of the switch."""
    try:
        return self._name.format(self.data['label'])
    except IndexError:
        return self._name
```